xml/System.Net.Sockets/Socket.xml

```diff
@@ -444,6 +444,13 @@ This method populates the <xref:System.Net.Sockets.Socket> instance with data ga
         <summary>Accepts an incoming connection.</summary>
         <returns>An asynchronous task that completes with the accepted Socket.</returns>
         <remarks>To be added.</remarks>
+        <exception cref="T:System.InvalidOperationException">An invalid operation was requested. This exception occurs if the accepting <see cref="T:System.Net.Sockets.Socket" /> is not listening for connections or the accepted socket is bound.
+
+ You must call the <see cref="M:System.Net.Sockets.Socket.Bind(System.Net.EndPoint)" /> and <see cref="M:System.Net.Sockets.Socket.Listen(System.Int32)" /> method before calling the <see cref="M:System.Net.Sockets.Socket.AcceptAsync(System.Net.Sockets.SocketAsyncEventArgs)" /> method.
+
+ This exception also occurs if the socket is already connected or a socket operation was already in progress using the specified <paramref name="e" /> parameter.</exception>
+        <exception cref="T:System.Net.Sockets.SocketException">An error occurred when attempting to access the socket.</exception>
```

Comment:
I believe we throw this only asynchronously from most (if not all) the methods. Do we document async exceptions similarly in other Networking areas?

Reply:
I think so, e.g. HttpClient has anotations for HttpRequestException on async methods (https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpclient.sendasync?view=net-7.0)

Reply:
same for AuthenticationException on SslStream.AuthenticateAs*Async (https://learn.microsoft.com/en-us/dotnet/api/system.net.security.sslstream.authenticateasclientasync?view=net-6.0)

Reply:
Looks like BCL is very inconsistent on this -- https://github.com/dotnet/dotnet-api-docs/issues/7840#issuecomment-1282761648, but at least you are making this consistent with the rest of the networking docs 😆